### DISCUSSION TITLE
Change button position and style

### DISCUSSION THREAD

**Discussion by gaiamolinaro** [jspsych/jsPsych#877] [Q&A] [answered]
I am trying to display two buttons using the jsPsych html-button-response plugin ([](https://www.jspsych.org/plugins/jspsych-html-button-response/)). I am using images as buttons instead of text. I specified them in my "choices:" option as `choices: ["<img src='" + ButtonLeft+ "'>", "<img src='" + ButtonRight+ "'>"],`.

By default, the buttons show up centered and below my stimulus. Instead, I would like to display them ON TOP of my stimulus (which is in turn composed of a bunch of overlapping images - a background + some items), one on the LEFT and the other one on the RIGHT, in specific positions relative to my background. The default style for the buttons is:

```
 .jspsych-btn {
  display: inline-block;
  padding: 6px 12px;
  margin: 0px;
  font-size: 14px;
  font-weight: 400;
  font-family: 'Open Sans', 'Arial', sans-serif;
  cursor: pointer;
  line-height: 1.4;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  background-image: none;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #333;
  background-color: #fff;
  border-color: #ccc;
}
```
How can I display the buttons in the way that I am trying to? Thank you so much in advance!

**Comment by vijaymarupudi** [comment, marked as answer]
I would not recommend solving this problem purely using CSS, as you will face problems when dealing with screens of different sizes.

I would recommend making a new plugin that satisfies these constraints. In a way.

You'll need to generate HTML that looks like this:

```html
<div id='button-container'>
  <img src="image-choice-one.png">
  <img src="image-src.png">
  <img src="image-choice-two.png">
</div>
```

And style it in css like this:

```css
#button-container {
  display: flex;
  justify-content: center;
}

#button-container > * {
  margin: 10px;
}
```

You'd then need to add [click listeners] to record which choice was clicked and then call `jsPsych.finishTrial()` with that data.

There is a tutorial on how to create a plugin from scratch that will help: <https://www.youtube.com/watch?v=XQcsFwAmbiw>

[click listeners]: https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event

**Comment by gaiamolinaro** [comment]
Changing the button_html parameter did it! But it is true that this is not the ideal solution. 

`button_html: ['<button class="jspsych-btn" style = "position:absolute; left:284.5px; top: 270px">%choice%</button>', '<button class="jspsych-btn" style = "position:absolute; right:284.5px; top: 270px">%choice%</button>']`

**Reply by vijaymarupudi** [reply]
I would be very wary of this solution. Try resizing your browser and you will find that stuff will not show up in the right place.

**Reply by jodeleeuw** [reply]
Following @vijaymarupudi, one thing you can do to is set an extra CSS rule for the `<div id="jspsych-content">` element.

```html
<style>
#jspsych-content { position: relative; }
</style>
```

This should make it easier to use `position: absolute` on elements that display within the jsPsych container, because when you put something in an absolute position, that position is relative to whichever nearest parent element is using relative positioning.

**Reply by gaiamolinaro** [reply]
Thank you both!
